src/topology/support.lean

```diff
@@ -0,0 +1,178 @@
+    has_compact_mul_support_def, exists_compact_superset_iff]
+
+@[to_additive]
+lemma has_compact_mul_support.intro [t2_space α] {K : set α}
+  (hK : is_compact K) (hfK : ∀ x ∉ K, f x = 1) : has_compact_mul_support f :=
+exists_compact_iff_has_compact_mul_support.mp ⟨K, hK, hfK⟩
+
+@[to_additive]
+lemma has_compact_mul_support.is_compact (hf : has_compact_mul_support f) :
+  is_compact (mul_tsupport f) :=
+hf
+
+@[to_additive]
+lemma has_compact_mul_support.mono' {f' : α → γ} (hf : has_compact_mul_support f)
+  (hff' : mul_support f' ⊆ mul_tsupport f) : has_compact_mul_support f' :=
+compact_of_is_closed_subset hf is_closed_closure $ closure_minimal hff' is_closed_closure
+
+@[to_additive]
+lemma has_compact_mul_support.mono {f' : α → γ} (hf : has_compact_mul_support f)
+  (hff' : mul_support f' ⊆ mul_support f) : has_compact_mul_support f' :=
+hf.mono' $ hff'.trans subset_closure
+
+@[to_additive]
+lemma has_compact_mul_support.comp_left (hf : has_compact_mul_support f) (hg : g 1 = 1) :
+  has_compact_mul_support (g ∘ f) :=
+hf.mono $ mul_support_comp_subset hg f
+
+@[to_additive]
+lemma has_compact_mul_support_comp_left (hg : ∀ {x}, g x = 1 ↔ x = 1) :
+  has_compact_mul_support (g ∘ f) ↔ has_compact_mul_support f :=
+by simp_rw [has_compact_mul_support_def, mul_support_comp_eq g @hg f]
+
+@[to_additive]
+lemma has_compact_mul_support.comp₂_left (hf : has_compact_mul_support f)
+  (hf₂ : has_compact_mul_support f₂) (hm : m 1 1 = 1) :
+  has_compact_mul_support (λ x, m (f x) (f₂ x)) :=
+begin
+  refine compact_of_is_closed_subset (hf.union hf₂) is_closed_closure _,
+  refine closure_minimal (λ x h2x, _) (is_closed_closure.union is_closed_closure) ,
+  refine union_subset_union subset_closure subset_closure _,
+  by_contra hx,
+  simp_rw [mem_union, not_or_distrib, nmem_mul_support] at hx,
+  apply h2x,
+  simp_rw [hx.1, hx.2, hm]
+end
+
+end
+
+section monoid
+
+variables [topological_space α] [monoid β]
+variables {f f' : α → β} {x : α}
+
+
+@[to_additive]
+lemma has_compact_mul_support.mul (hf : has_compact_mul_support f)
+  (hf' : has_compact_mul_support f') : has_compact_mul_support (f * f') :=
+by apply hf.comp₂_left hf' (mul_one 1) -- `by apply` speeds up elaboration
+
+end monoid
+
+section monoid
+
+variables [topological_space α] [monoid_with_zero β] [add_monoid γ] [distrib_mul_action β γ]
+variables {f : α → β} {f' : α → γ} {x : α}
+
+
+lemma has_compact_support.smul (hf : has_compact_support f)
+  (hf' : has_compact_support f') : has_compact_support (f • f') :=
+by apply hf.comp₂_left hf' (smul_zero 0) -- `by apply` speeds up elaboration
+
+end monoid
+
+section monoid_with_zero
+
+variables [topological_space α] [mul_zero_class β]
+variables {f f' : α → β} {x : α}
+
+@[to_additive]
+lemma has_compact_support.mul (hf : has_compact_support f)
+  (hf' : has_compact_support f') : has_compact_support (f * f') :=
+by apply hf.comp₂_left hf' (mul_zero 0) -- `by apply` speeds up elaboration
+
```

Comment:
I miss the lemma in groups with zero that, if `f` has compact (additive) support and `f'` is arbitrary, then `f \bu f'` has compact support (and friends).